extensions/bear/src/index.tsx

```diff
@@ -25,6 +26,7 @@ export default function SearchNotes(props: LaunchProps<{ arguments: SearchNotesA
   }
 
   const showDetail = (notes ?? []).length > 0 && getPreferenceValues().showPreviewInListView;
+  const handleTagChange = (tag: string) => setNotes(db?.getNotes(tag));
```

Comment:
I think you'll want the `#` prefix / suffix to filter _only_ by the tag, not just for all occurrences of the tag's name. E.g.:

```suggestion
  const handleTagChange = (tag: string) => {
    const query = tag.includes(" ") ? `#${tag}#` : `#${tag}`;
    setNotes(db?.getNotes(query));
  }
```

Reply:
Additionally, calling `setNotes` here will mean that when the user modifies the search, the tag filter will no longer apply.

One solution would be to track the tag filter in a new `useState` variable, add that to the dependency array for the `useEffect` call further up this component, then modify the `getNotes` function to accept a tag that we inner join against. Unfortunately simply appending the tag to the existing query won't work because we match the entire query string rather than tokenizing the query (see https://github.com/raycast/extensions/issues/2710).

Alternatively we could just disable the search box when you're filtering, but that's not ideal!

Reply:
Hey @hmarr! Thanks for taking a look and spotting some issues. I'm fixing all of them right now. The only problem that I have is that the when no tag is selected, let's say, in the `useEffect` at the beginning, instead of showing me all the notes, it shows me the untagged notes.

This is the modified version of the getNotes 
```ts
  getNotes(searchQuery: string, tag?: string): Note[] {
    const statement = this.database.prepare(
      this.getBearVersion() === 2 ? BearDb.searchNotesQueries.v2 : BearDb.searchNotesQueries.v1
    );
    if (tag) {
      statement.bind({ ":query": `${searchQuery}`, ":tag": `${tag}` });
    } else {
      statement.bind({ ":query": `${searchQuery}` });
    }
    const results: Note[] = [];
    while (statement.step()) {
      const row = statement.getAsObject();
      results.push(this.toNote(row));
    }

    statement.free();
    return results;
  }
```
And this is the modified query 
```sql
SELECT
  notes.ZUNIQUEIDENTIFIER AS id,
  notes.ZTITLE AS title,
  notes.ZTEXT AS text,
  notes.ZMODIFICATIONDATE AS modified_at,
  notes.ZCREATIONDATE AS created_at,
  group_concat(tags.ZTITLE) AS tags,
  notes.ZENCRYPTED AS encrypted
FROM
  ZSFNOTE AS notes
LEFT OUTER JOIN
  Z_5TAGS AS notes_to_tags ON notes.Z_PK = notes_to_tags.Z_5NOTES
LEFT OUTER JOIN
  ZSFNOTETAG AS tags ON notes_to_tags.Z_13TAGS = tags.Z_PK
WHERE
  -- When there is a query, filter the body by that query, otherwise
  -- ignore the query
  (
    lower(notes.ZTITLE) LIKE lower('%' || :query || '%')
    OR lower(notes.ZTEXT) LIKE lower('%' || :query || '%')
    OR :query = ''
    OR lower(notes.ZUNIQUEIDENTIFIER) LIKE lower('%' || :query || '%')
  )
  -- When there is a tag query, filter the tags by that query, otherwise
  -- ignore the tag query
  AND (:tag = '' OR tags.ZTITLE LIKE lower('%' || :tag || '%'))  
  -- Ignore trashed, archived, and empty notes
  AND notes.ZARCHIVED = 0
  AND notes.ZTRASHED = 0
  AND (
    notes.ZTEXT IS NOT NULL
    OR notes.ZENCRYPTED = 1
  )
GROUP BY
  notes.ZUNIQUEIDENTIFIER
ORDER BY
  -- Sort title matches ahead of body matches
  CASE WHEN (
    lower(notes.ZTITLE) like lower('%' || :query || '%')
    OR :query = ''
  ) THEN 0 ELSE 1 END,
  -- When there are multiple title matches, sort by last modified
  notes.ZMODIFICATIONDATE DESC
LIMIT
  20
```
I don't know if there is any solution to this query or we just need to make a new query for this situation

Reply:
Could it be that you're not passing the `:tag` parameter in at all if it's `undefined`, meaning it won't be equal to the empty string? If that's not it, I'm happy to take a look at locally if you want to push an update to the branch.

Additionally, I just realised that if we filter by tag using the `WHERE` clause, we'll only include the tag we're filtering by in the `group_concat(tags.ZTITLE)` column, which means the note preview won't show other tags the note has. Given this, it might be better to move the tag filter to a `HAVING` clause, or to use a CTE to separate the joining on tags from the tag filtering?